web/packages/teleterm/src/ui/services/workspacesService/accessRequestsService.ts

```diff
@@ -88,13 +90,20 @@ export class AccessRequestsService {
         };
       }
 
+      const requestsArray = Array.isArray(requestedResources)
+        ? requestedResources
+        : [requestedResources];
+
       const { resources } = draftState.pending;
+      const allAdded = requestsArray.every(r => resources.has(r.resource.uri));
 
-      if (resources.has(request.resource.uri)) {
-        resources.delete(request.resource.uri);
-      } else {
-        resources.set(request.resource.uri, getRequiredProperties(request));
-      }
+      requestsArray.forEach(request => {
+        if (allAdded) {
```

Comment:
This is a behavior where either all items are added or all items are removed seems worth adding a test for. I feel like it'd be easy to forget about this behavior when refactoring stuff.